src/haddock/modules/sampling/lightdock/__init__.py

```diff
@@ -38,18 +38,18 @@ def confirm_installation(cls):
     def _run(self):
         """Execute module."""
         # Get the models generated in previous step
-        models_to_score = [
+        self.output_models = [
```

Comment:
Would it not make more sense to use a local variable here, now you load all input models in a variable called `output_models`. And especially at line 48 to check if there are multiple input models looks weird. If it was output you would just have made sure that it was only one model when you generated the output.